application/models/gateway/Gammu_model.php

```diff
@@ -52,19 +52,35 @@ function _send_wap_link($data)
 				&& $data['type'] === 'waplink')
 		{
 			$cmd = '"'.$this->config->item('gammu_sms_inject').'"' .' -c ' .'"'.$this->config->item('gammu_config').'"'. ' WAPINDICATOR ' . $data['dest']. ' "'. $data['url']. '"  "'.$data['message'].'" ';
-			$ret = exec ($cmd);
-			preg_match('/with ID ([\d]+)/i', $ret, $matches);
-			$insert_id = $matches[1];
-			if (empty($insert_id))
+			$ret = exec ($cmd, $output, $result_code);
+			if ($result_code !== 0)
 			{
-				die('<div class="notif" style="color:red">'.tr('Could not send message. Make sure Gammu path is correctly set.').'</div>');
```

Comment:
This `die` was removed intentionally ?  In that case the function should exit after here.

Reply:
Yes, the call to show_error just above replace it.